### ISSUE TITLE
CVE-2023-5129

### ISSUE THREAD

**Issue by Speedy1991** [thumbor/thumbor#1591]
Is thumbor affected by CVE-2023-5129?

**Comment by genisd** [comment]
Thumbor in docker uses debian bullseye as build image.
Which installs/uses `libwebp-dev` to the best of my knowledge, which is affected.

Updates can be pulled from debian https://security-tracker.debian.org/tracker/CVE-2023-5129

**Comment by Speedy1991** [comment]
The [Pillow 9.*](https://github.com/thumbor/thumbor/blob/master/setup.py#L155) dependency is also affected but fixed [with 10.0.1](https://github.com/python-pillow/Pillow/issues/7425)

**Comment by genisd** [comment]
Thank you Speedy1991.
I just double checked, and indeed on the docker rebuild that fix was pulled in for us also :heart:

**Comment by anselm-helbig** [comment]
I was using a proof-of-concept webp image to test Thumbor's vulnerability, following the instructions at [Ben Hawkes' blog](https://blog.isosceles.com/the-webp-0day/). When served by Thumbor, it caused the decoder to crash like this:
```
2023-09-29 15:36:26 thumbor:ERROR ERROR: Traceback (most recent call last):
  File "/home/apprunner/.local/lib/python3.11/site-packages/thumbor/handlers/__init__.py", line 214, in get_image
    result = await self._fetch(self.context.request.image_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apprunner/.local/lib/python3.11/site-packages/thumbor/handlers/__init__.py", line 956, in _fetch
    raise fetch_result.exception
  File "/home/apprunner/.local/lib/python3.11/site-packages/thumbor/handlers/__init__.py", line 924, in _fetch
    self.context.request.engine.load(fetch_result.buffer, extension)
  File "/home/apprunner/.local/lib/python3.11/site-packages/thumbor/engines/__init__.py", line 201, in load
    image_or_frames = self.create_image(buffer)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apprunner/.local/lib/python3.11/site-packages/thumbor/engines/pil.py", line 109, in create_image
    img = Image.open(BytesIO(buffer))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apprunner/.local/lib/python3.11/site-packages/PIL/Image.py", line 3282, in open
    im = _open_core(
         ^^^^^^^^^^^
  File "/home/apprunner/.local/lib/python3.11/site-packages/PIL/Image.py", line 3263, in _open_core
    im = factory(fp, filename)
         ^^^^^^^^^^^^^^^^^^^^^
  File "/home/apprunner/.local/lib/python3.11/site-packages/PIL/ImageFile.py", line 117, in __init__
    self._open()
  File "/home/apprunner/.local/lib/python3.11/site-packages/PIL/WebPImagePlugin.py", line 62, in _open
    self._decoder = _webp.WebPAnimDecoder(self.fp.read())
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: could not create decoder object

2023-09-29 15:36:26 thumbor:ERROR [BaseHandler] get_image failed for url `development/666.webp`. error: `could not create decoder object`
```

Thumbor uses Pillow which ships with libwebp. Version 10.0.1 has a fix, but Thumbor still requires 9.*. You can [build Pillow from source](https://pillow.readthedocs.io/en/latest/installation.html#building-from-source), then it should use the OS' libraries. I couldn't get this to work with our deployment images yet.

**Comment by Speedy1991** [comment]
Who is the maintainer to ping in this repo?
This is a hard security concern and should be fixed asap IMO

Also some updates in the docs are required, because there is no email or mailing list to report security stuff without opening an issue. Or at least i do not dind any Informationen about this

**Comment by marcelometal** [comment]
https://github.com/thumbor/thumbor/pull/1592

**Comment by RaphaelVRossi** [comment]
We will have to drop python 3.7 for that.

https://github.com/python-pillow/Pillow/issues/7195

**Comment by devppjr** [comment]
closing this issue, thanks everyone! 😄

**Comment by RaphaelVRossi** [comment]
Fix was released [7.6.0](https://github.com/thumbor/thumbor/releases/tag/7.6.0) 🚀
